server/src/__tests__/embedded.test.ts

```diff
@@ -0,0 +1,109 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import fs from 'fs'
+import path from 'path'
+
+import { embeddedDocumentsManager } from '../embedded-languages/documents-manager'
+import { generateEmbeddedDocuments, getEmbeddedDocumentUriStringOnPosition } from '../embedded-languages/general-support'
+import { analyzer } from '../tree-sitter/analyzer'
+import { generateParser } from '../tree-sitter/parser'
+import { FIXTURE_DOCUMENT } from './fixtures/fixtures'
+
+describe('Embedded Documents', () => {
```

Comment:
the term "embedded" applies also to yocto targets. you could make the code more clear by always using "embedded language" instead of "embedded document"

Reply:
I really have naming problems here. The most accurate I can think about would be EmbeddedLanguageDocument, but it would be way too long (the current naming is already much too long). I would also hate to use a made up acronym such as "ELD"

Renaming EmbeddedDocumentManager as EmbeddedLanguageManager would be confusing, since it is managing documents, not the "embedded languages".

Renaming generateEmbeddedDocuments as generateEmbeddedLanguage would also be confusing, since it is generating documents, not "embedded languages".

I still could do it, but I hope we can think of something better 😬

Reply:
I have no problems with longer names when they are justified personnally. EmbeddedLanguageDoc?

Reply:
Done, we use EmbeddedLanguageDoc